addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/service/SmappeeService.java

```diff
@@ -0,0 +1,294 @@
+                    readingsUpdate.invalidConfig();
+                }
+            }
+        };
+
+        scheduledJob = scheduledExecutorService.scheduleAtFixedRate(runnable, 0, this.refreshInterval,
+                TimeUnit.MILLISECONDS);
+    }
+
+    ScheduledFuture<?> scheduledJob;
+
+    public void stopAutomaticRefresh() {
+        scheduledJob.cancel(true);
+    }
+
+    public SmappeeDeviceReading getDeviceReadings() throws InvalidConfigurationException {
+        int currentTry = 0;
+        do {
+            try {
+                DateTime nowUtc = DateTime.now(DateTimeZone.UTC).minusMinutes(1);
+                DateTime nowUtcMinus20Min = DateTime.now(DateTimeZone.UTC).minusMinutes(20);
+
+                String nowUtcMillis = String.valueOf(nowUtc.getMillis());
+                String nowUtcMinus20MinMillis = String.valueOf(nowUtcMinus20Min.getMillis());
+
+                // https://app1pub.smappee.net/dev/v1/servicelocation/123/consumption?aggregation=1&from=1388534400000&to=1391212800000
+
+                String responseReadings = GetData("/dev/v1/servicelocation/" + this.serviceLocationId
+                        + "/consumption?aggregation=1&from=" + nowUtcMinus20MinMillis + "&to=" + nowUtcMillis);
+
+                SmappeeDeviceReading readings = gson.fromJson(responseReadings, SmappeeDeviceReading.class);
+
+                // Sum of 5 minutes in Wh so average power is 60/5 x
+
+                for (SmappeeDeviceReadingConsumption consumption : readings.consumptions) {
+                    consumption.consumption = consumption.consumption * 12;
+                    consumption.solar = consumption.solar * 12;
+                }
+
+                logger.debug("smappee'{}' read", new Object[] { this.serviceLocationId });
+
+                return readings;
+
+            } catch (Exception se) {
+                logger.debug("failed to read smappee '{}'", new Object[] { se.getMessage() });
+            }
+        } while (currentTry++ < this.retry);
+
+        return null;
+
+    }
+
+    public boolean initialize() {
+        // get service locations
+        String response = GetData("/dev/v1/servicelocation");
+
+        if (response.isEmpty()) {
+            return false;
+        }
+
+        SmappeeServiceLocationResponse smappeeServiceLocationResponse = gson.fromJson(response,
+                SmappeeServiceLocationResponse.class);
+
+        if (smappeeServiceLocationResponse == null) {
+            return false;
+        }
+
+        for (SmappeeServiceLocation smappeeServiceLocation : smappeeServiceLocationResponse.serviceLocations) {
+            if (smappeeServiceLocation.name.equals(this.serviceLocationName)) {
+                this.serviceLocationId = Integer.toString(smappeeServiceLocation.serviceLocationId);
+                isInitialized = true;
+
+                return true;
+
+            }
+        }
+        return false;
+    }
+
+    public boolean isInitialized() {
+        return isInitialized;
```

Comment:
boolean variable should be called initialized the action(method) is called isInitialized.

Reply:
ok